sensitivity/src/main/java/powsybl/tutorials/sensitivity/SensitivityTutorial.java

```diff
@@ -0,0 +1,199 @@
+        // If the file doesn't exist, create it
+        if (!jsonTwtResultFile.exists()) {
+            boolean fileCreated = jsonTwtResultFile.createNewFile();
+            if (!fileCreated) {
+                throw new IOException("Unable to create the result file");
+            }
+        }
+        SensitivityComputationResultExporter exporter = new JsonSensitivityComputationResultExporter();
+        try (FileOutputStream os = new FileOutputStream(jsonTwtResultFile.toString())) {
+            exporter.export(sensitivityTwtComputationResults, new OutputStreamWriter(os));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        // 7. Perform a systematic sensitivity analysis
+        // TODO: implement the API for the Hades2 systematic sensitivity studies
+        // First, implement a contingencies provider.
+        // Here the list of contingencies is composed of the lines that are not monitored
+        // in the sensitivity analysis.
+        ContingenciesProvider contingenciesProvider = n -> n.getLineStream()
+                .filter(l -> {
+                    final boolean[] isContingency = {true};
+                    monitoredLines.forEach(monitoredLine -> {
+                        if (l.equals(monitoredLine)) {
+                            isContingency[0] = false;
+                            return;
+                        }
+                    });
+                    return isContingency[0];
+                })
+                .map(l -> new Contingency(l.getId(), new BranchContingency(l.getId())))
+                .collect(Collectors.toList());
+        // This makes a total of 11 contingencies
+        LOGGER.info("Number of contingencies: {}", contingenciesProvider.getContingencies(network).size());
+
+        // For each contingency, create a variant of the network and run a sensitivity analysis.
+        contingenciesProvider.getContingencies(network).forEach(cont -> {
+            // Create a new variant and set it as the WorkingVariant.
+            network.getVariantManager().cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID,
+                    CONTINGENCY_VARIANT);
+            network.getVariantManager().setWorkingVariant(CONTINGENCY_VARIANT);
+            // Apply the contingency on the network in that variant.
+            cont.toTask().modify(network, computationManager);
+
+            // Run the sensitivity computation with respect to the PST tap position on that network.
+            SensitivityComputationResults contingencyResults = sensitivityComputation.run(
+                    twtFactorsProvider, CONTINGENCY_VARIANT,
+                    SensitivityComputationParameters.load()).join();
+
+            // Print the sensitivity values in the terminal.
+            LOGGER.info("Contingency {}:", cont.getId());
+            contingencyResults.getSensitivityValues().forEach(value ->
```

Comment:
Maybe a more understandable logger ? We print:
````
main] INFO powsybl.tutorials.sensitivity.SensitivityTutorial - Contingency BBE1AA1  BBE2AA1  1:
[main] INFO powsybl.tutorials.sensitivity.SensitivityTutorial - Value: 39.78589630126953
[main] INFO powsybl.tutorials.sensitivity.SensitivityTutorial - Value: 39.78589630126953
````
In the web site, 39.7858.... is the partial derivative and could be interpreted as the flow increasing of 39MW on the monitored branch flow if we move the current tap of the PST to the next one. Am I right ? Could you correct if I am not ? By next one, it is necessary a positive increase ?

Reply:
I modified the comments in order to better explain this. Please let me know if it is still unclear.